lib/raygun/javascript_tracker.rb

```diff
@@ -0,0 +1,46 @@
+# Client for injecting JavaScript code for tracking front end exceptions
+# https://raygun.com/docs/languages/javascript
+module Raygun
+  class JavaScriptTracker
+    def head_html
+      return unless js_api_key?
+      [
+        '<script type="text/javascript">',
+        '!function(a,b,c,d,e,f,g,h){a.RaygunObject=e,a[e]=a[e]||function(){',
+        '(a[e].o=a[e].o||[]).push(arguments)},f=b.createElement(c),g=b.getElementsByTagName(c)[0],',
+        'f.async=1,f.src=d,g.parentNode.insertBefore(f,g),h=a.onerror,a.onerror=function(b,c,d,f,g){',
+        'h&&h(b,c,d,f,g),g||(g=new Error(b)),a[e].q=a[e].q||[],a[e].q.push({',
+        'e:g})}}(window,document,"script","//cdn.raygun.io/raygun4js/raygun.min.js","rg4js");',
+        '</script>'
+      ].join('').html_safe
+    end
+
+    def body_html
+      return unless js_api_key?
+      [
+        '<script type="text/javascript">',
+        "rg4js('apiKey', '#{js_api_key}');",
+        "rg4js('enableCrashReporting', true);",
+        user_context,
+        '</script>'
+      ].join('').html_safe
+    end
+
+    private
+
+    def user_context
+      # TODO, Pull in user data from somewhere.
```

Comment:
Yes, you should be able to just pull the hash out of the ENV and pass it in as `rg4js('setUser', ...)`. I'm not exactly sure how you can convert it to a JS object but shouldn't be too hard. The hash fields are already mapped to the same fields Raygun4JS expects.

Reply:
I decided to leave this for another PR, I think users who are interested in the setting their user context should be ok to read the docs (Plus I'm keen to get this in ;) )